WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormTableViewDataSource.swift

```diff
@@ -121,7 +119,17 @@ private extension ProductFormTableViewDataSource {
         else {
             cell.configure(with: productImageStatuses, config: .extendedAddImages, productUIImageLoader: productUIImageLoader)
         }
-
+        if productLabel.contains("draft") {
+            cell.productStatusLabelHolder.backgroundColor = UIColor.gray
+            cell.productStatusLabelHolder.layer.cornerRadius = CGFloat(2.0)
+            cell.productStatusLabelHolder.layer.borderWidth = CGFloat(0.5)
+            let statusLabel = NSLocalizedString("Draft", comment: "Display label for the product's draft status")
+            cell.productStatusLabel.text? = statusLabel
+            cell.productStatusLabel.font = UIFont.preferredFont(forTextStyle: .body)
+            cell.productStatusLabel.textColor = UIColor.white
+        } else {
+            cell.productStatusLabel.text = ""
```

Comment:
This isn't enough to clear the badge. When you are viewing a draft product and hit publish, the badge will clear the text but leave the gray box. If you leave the screen and come back (or visit a published product), there will be a white box on top of the first image:

![Screen Shot 2021-08-09 at 17 30 35](https://user-images.githubusercontent.com/8739/128732680-b71eaa9b-7afb-44cc-986c-e24d75ec9bf0.png)
![Screen Shot 2021-08-09 at 17 30 38](https://user-images.githubusercontent.com/8739/128732687-88be7095-190b-4c66-8a7e-7bff4aa3741e.png)

Reply:
Gotcha! I missed this when testing 😓